sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_deployment.py

```diff
@@ -212,6 +217,14 @@ def _to_rest_object(self, location: str) -> BatchDeploymentData:  # pylint: disa
             properties=self.properties,
         )
 
+        if is_private_preview_enabled() and self.job_definition:
+            non_flat_data = {}
+            non_flat_data["component_deployment"] = self.job_definition._to_dict()
```

Comment:
Should this be "job_definition" over component_deployment? Or are these synonymous terms?